crypto/types/src/seal.rs

```diff
@@ -0,0 +1,54 @@
+use libp2p_identity::PeerId;
+
+use crate::errors::CryptoError;
+use crate::keypairs::OffchainKeypair;
+use crate::types::OffchainPublicKey;
+
+/// Performs randomized encryption of the given serializable object, so that
+/// only the recipient with the given `peer_id` can [decrypt it](unseal_data).
+pub fn seal_data<T: serde::Serialize>(data: T, peer_id: PeerId) -> crate::errors::Result<Box<[u8]>> {
```

Comment:
Is this backwards compatible with 2.1?

Reply:
This feature wasn't in 2.1, so there's nothing to be compatible with.